### PR TITLE
[NO-ISSUE]updating readme for beta snapshot requirement

### PR DESCRIPTION
**Pull request by david-enli**
updating CRD/controller requirement for Snapshot BETA
some other minor change to reflect recent changes.

### CODE REVIEW

File: pureStorageDriver/README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ This helm chart installs the Pure Service Orchestrator CSI plugin on a Kubernete
   - Ubuntu 18.04
 - #### Environments Supported*:
   - Kubernetes 1.13+
-    - NOTE: for Kubernetes 1.17, there is an [issue](https://github.com/kubernetes/kubernetes/issues/87852) using vxlan with Flannel or Calico 
+    - NOTE: for Kubernetes 1.17, there is an [issue](https://github.com/kubernetes/kubernetes/issues/87852) using vxlan with Flannel or Calico, the issue has been fixed after 18.5
```
Comment (sdodsley) [inline pureStorageDriver/README.md:20]: 1.18.5

File: pureStorageDriver/README.md [hunk 2]
```diff
@@ -169,6 +169,8 @@ volumesnapshotclasses.snapshot.storage.k8s.io    2019-11-21T17:25:23Z
 volumesnapshotcontents.snapshot.storage.k8s.io   2019-11-21T17:25:23Z
 volumesnapshots.snapshot.storage.k8s.io          2019-11-21T17:25:23Z
 ```
+Note that PSO CSI has upgraded its sidecars to support BETA version snapshotter APIs. The snapshotter CRDs for Beta version APIs have been upgraded, make sure for 1.17 and later versions, apply release-2.0 CRDs as shown
```
Comment (sdodsley) [inline pureStorageDriver/README.md:172]: I think this whole added section needs to be more prominent - it is crucial that users are aware of this to stop unnecessary escallations

File: pureStorageDriver/README.md (hunk 1 above)
Comment (pure-jliao) [reply pureStorageDriver/README.md:20]: it's fixed in 1.18.6? or 1.18.5, it's a little bit confusing.

Comment (david-enli) [reply pureStorageDriver/README.md:20]: I know, the change log says 1.18.6 but I tested 1.18.5 and it was already working, and looks like the actual PR was merged before 18.5 release

File: pureStorageDriver/README.md (hunk 2 above)
Comment (david-enli) [reply pureStorageDriver/README.md:172]: will do

**Pull request closed without merging**